### PR TITLE
refactor(vm-runner): Improve VM runner / VM playground

### PR DESCRIPTION
**Pull request by slowli**
## What ❔

Various minor improvements to VM runner / VM playground:

- Get batch storage asynchronously, so that it works efficiently with snapshot storage.
- Add metrics / logs to ensure that snapshot storage works as expected.

## Why ❔

Improves usability.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: core/node/vm_runner/src/process.rs [hunk 1]
```diff
@@ -148,39 +177,16 @@ impl VmRunner {
...
-
-            self.io
-                .mark_l1_batch_as_processing(&mut self.pool.connection().await?, next_batch)
-                .await?;
-            let handle = tokio::task::spawn(Self::process_batch(
-                batch_executor,
-                batch_data.l2_blocks,
-                output_handler,
-            ));
+            let handle = tokio::spawn(self.clone().process_batch(next_batch));
```
Comment (slowli) [inline core/node/vm_runner/src/process.rs:188]: I'm not 100% sure this change is correct for existing VM runner components other than the VM playground. With this async pipeline, it means that `mark_l1_batch_as_processing()` can be called out of order (e.g., first for batch 5, then 3, then 4).

Comment (itegulov) [reply core/node/vm_runner/src/process.rs:188]: The order of `mark_l1_batch_as_processing` shouldn't matter as long as `mark_l1_batch_as_completed` is sequential. All existing implementations use processing stage purely for metrics/observability. Maybe we should reflect that the implementation should not rely on order of execution in `mark_l1_batch_as_processing`'s rustdoc

Comment (itegulov) [reply core/node/vm_runner/src/process.rs:188]: Ah, sorry, I skimmed through the code somehow missed that

**Review by itegulov — APPROVED** [review]
Big fan of the changes to VM runner!

**Pull request merged**